README.md

```diff
@@ -1,2 +1,149 @@
+- To start developing the critical ability to identify distinct elements of your app and their relations.
+- To apply this knowledge, and learn how you might begin architecting an app pre-build.
+
+### What Is Software Architecture Design?
+
+<iframe src="https://giphy.com/embed/85O5hFDD2xWeY" width="480" height="274" frameBorder="0" class="giphy-embed" allowFullScreen></iframe><p><a href="https://giphy.com/gifs/software-85O5hFDD2xWeY"></a></p>
+
+##### Architecture is the high-level / overall structure of a software system.
+
+It is like showing a route on a map. Both the route and the map display are abstractions. They don't exist in real life, but they allow us to find our way and conceptualise the landscape.
+
+<iframe src="https://giphy.com/embed/kxhY6bj6JA7Qs" width="480" height="239" frameBorder="0" class="giphy-embed" allowFullScreen></iframe><p><a href="https://giphy.com/gifs/world-satisfying-map-kxhY6bj6JA7Qs"></a></p>
+
+
+##### Architecture shows how all the parts of an app might interact with each other.
+
+We can take the example of making an API call from the front end.
+
+![](http://www.dnbpartner.com/wp-content/uploads/2015/02/an_api_call.jpg)
+
+- Our user interacts with the graphic interface, maybe they click a button with an event listener attached to it.
+- This triggers an instance of an API call.
+- The API call sends a request to the database server, which returns a response.
+- The response is a big object which needs to be filtered through.
+- The desired results are sent to the DOM, updating the display.
+- The user interacts with the updated graphic interface.
+
+_The cycle continues..._
+
+Let's draw how this might appear on the board.
+
+## Types of Software Architecture
+
+There are [many different ways to design a map](https://www.thoughtco.com/types-of-maps-1435689), depending on its purpose. You get road maps, political maps, climate maps, topographic maps... maps provide a very interesting insight into our ability to work with abstract information.
+
+Similarly, there are many different approaches to architecting software, depending on the situation.
+
+For a comprehensive overview of different Architectural Styles and Patterns, have a cheeky look at [this guide](https://msdn.microsoft.com/en-us/library/ee658117.aspx). But don't get overwhelmed; every organisation you work with will have a unique approach to architecture that you will learn on the go.
+
+Let's look at some examples of common architectural styles.
+
+We'll split into our project groups and tackle one style per group.
+
+| Style        | Description           | Example Diagram  |
+| ------------- |:-------------:| -----:|
+| Client/Server      | Segregates the system into two applications, where the client makes requests to the server. In many cases, the server is a database with application logic represented as stored procedures. | ![](http://4.bp.blogspot.com/-U7IWBX8Od2E/UVQsx7PuVyI/AAAAAAAAABU/m5ckgXBLl-A/s1600/cc.jpg) |
```

Comment:
Good points. I actually think it might be best to remove this section completely now, because it isn't in line with the learning outcomes.